src/components/Contacts/ContactDetails/ContactDetails.tsx

```diff
@@ -0,0 +1,43 @@
+import { AppBar, Box, Tab, Tabs } from '@material-ui/core';
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useGetContactDetailsLazyQuery } from './ContactDetails.generated';
+import { ContactDetailsHeader } from './ContactDetailsHeader/ContactDetailsHeader';
+
+interface Props {
+  accountListId: string;
+  contactId: string | null;
+}
+
+export const ContactDetails: React.FC<Props> = ({
+  accountListId,
+  contactId,
+}: Props) => {
+  const { t } = useTranslation();
+
+  const [
+    loadContactDetails,
+    { data, loading },
+  ] = useGetContactDetailsLazyQuery();
+
+  useEffect(() => {
+    if (contactId != null) {
+      loadContactDetails({ variables: { accountListId, contactId } });
+    }
+  }, [contactId]);
```

Comment:
I think the header should load data immediately. Depending on how the tabs work, they should lazy load if they render immediately but should load data immediately if they aren't rendered until navigated to.

`contactId` shouldn't be nullable but if was needed to be, `skip` could be used in this case. Tatai was using this pattern of lazy load with a `useEffect` instead of a `skip` in some places. I'm working on changing those in my ts-strict PR.

Reply:
Yeah the way I'm handling the drawer opening/closing is that the the ContactDetails is rendered along with the whole Contact screen, but is controlled by `hidden`.  When the user first launches the screen, there is no contactId passed to ContactDetails, and I don't want to call the query until we have a contactId to query by, so a lazy query seemed like the best idea.  Later on, the user clicks on a row, the contactId is passed to the ContactDetails that is already present, the `useEffect` detects a change of value and launches the query.  It worked pretty well when I was testing it, opening and closing the ContactDetails for different rows.  But I'm not too attached to this implementation, if you know of a better way I'm open to trying it.

Reply:
I think it would be better to not render the details panel at all when it's not shown instead of using hidden to control it. Then you can just query as normal in the child component when it mounts.